pandas/tests/test_frame.py

```diff
@@ -6320,6 +6319,20 @@ def test_to_csv_from_csv(self):
             xp.columns = lmap(int,xp.columns)
             assert_frame_equal(xp,rs)
 
+        with ensure_clean() as path:
+            # GH 10833 (TimedeltaIndex formatting)
+            dt = pd.Timedelta(seconds=1)
+            df_orig = pd.DataFrame({'data': list(range(10))},
+                                   index=[i*dt for i in range(10)])
+            df_orig.index.rename('timestamp', inplace=True)
```

Comment:
I have the same result as you on your piece of code.

Take a loot at this example:

    path = 'tempfile'
    dt = pd.Timedelta(seconds=1)
    df = pd.DataFrame({'dt_data': [i*dt for i in range(3)]},
                      index=pd.Index([i*dt for i in range(3)],
                                  name='dt_index'))
    df.to_csv(path)

    result = pd.read_csv(path, index_col='dt_index')
    print(result.index)
    >>> Index(['00:00:00', '00:00:01', '00:00:02'], dtype='object', name='dt_index')
    result.index = pd.to_timedelta(result.index)
    print(result.index)
    >>> TimedeltaIndex(['00:00:00', '00:00:01', '00:00:02'], dtype='timedelta64[ns]', freq=None)

Seems like it disappears after `Index` -> `TimedeltaIndex` conversion.